h2o-persist-s3/src/main/java/water/persist/PersistS3.java

```diff
@@ -95,8 +95,11 @@ public H2OAWSCredentialsProviderChain() {
     public AWSCredentials getCredentials() {
       final IcedS3Credentials s3Credentials = DKV.getGet(IcedS3Credentials.S3_CREDENTIALS_DKV_KEY);
 
-      if (s3Credentials != null && s3Credentials._secretKeyId != null && s3Credentials._secretAccessKey != null) {
```

Comment:
removed check `s3Credentials != null` on purspose?

Reply:
No, this definitely has to be there !

Reply:
Added. Thank you for catching this.